c/collections.c

```diff
@@ -1417,6 +1417,84 @@ void *qRemove(Queue *q){
 #endif /* END OF OS-VARIANT Queue stuff */
 
 
+/* The Array List (flexible Array thing that smells like Java and javascript) */
+
+static void *arrayListAlloc(ArrayList *list, uint32_t size, char *location){
+  if (list->slh){
+    return (void*)SLHAlloc(list->slh,size);
+  } else{
+    return safeMalloc(size,location);
+  }
+}
+
+ArrayList *makeArrayList(){
```

Comment:
It's better to use `void` for function which don't have arguments in C, otherwise the caller can pass whatever and the compiler won't complain.

Reply:
You are right and I am guilty of this sin in many places.  I fixed this and a few others.  But not all

recovery.h:bool recoveryIsRouterEstablished();
recovery.h:void recoveryPop();
recovery.h:void recoveryEnableCurrentState();
recovery.h:void recoveryDisableCurrentState();
recovery.h:int recoveryRemoveRouter();
recovery.h:bool recoveryIsRouterEstablished();
recovery.h:RecoveryContext *getRecoveryContext();
recovery.h:  RecoveryContext *context = getRecoveryContext(); \
recovery.h:void recoveryPop();
recovery.h:void recoveryEnableCurrentState();
recovery.h:void recoveryDisableCurrentState();
recovery.h:extern RecoveryContext *rcvrgcxt();